collect_app/src/main/java/org/odk/collect/android/formmanagement/FormDeleter.java

```diff
@@ -29,5 +33,7 @@ public void delete(Long id) {
         } else {
             formsRepository.softDelete(form.getId());
         }
+
+        fastExternalItemsetsRepository.deleteAllByCsvPath(form.getFormMediaPath() + File.separator + "itemsets.csv");
```

Comment:
It felt cleaner to avoid having the `FormsRepository` care about itemsets and have the clean up logic here.